test/component-test.js

```diff
@@ -35,8 +35,17 @@ describe('component creation', () => {
     });
 
     context('node is scoped', () => {
-      it('applies text content to element');
-      it('recursively creates elements with scoped data');
+      it('applies text content to element', () => {
+        var attr = new Attribute('data-prop', 'lol');
+        var node = new Node('div', [attr]);
+        var data = {lol: 'wat'};
+
+        var {tag, props, content} = createComponent(node, data);
```

Comment:
hmm, so in this case it's sort of being used as multiple return values. I reads a lot like a tuple from FP which is probably why I ending up reaching for it... HMM